packages/terra-tabs/src/common-tabs/_Tabs.jsx

```diff
@@ -336,30 +385,83 @@ class Tabs extends React.Component {
+      }));
+      this.setState({ visibleTabData: updatedTabData }, () => {
+        this.setIsOpen(false);
+        onSelect(itemKey, metaData);
+      });
     };
   }
 
   wrapOnSelectHidden(onSelect) {
     return (itemKey, metaData) => {
       if (this.isOpen) {
-        onSelect(itemKey, metaData);
+        const updatedTabData = this.state.visibleTabData.map((tab) => ({
+          ...tab,
+          isSelected: tab.id === itemKey,
+        }));
+        this.setState({ visibleTabData: updatedTabData }, () => {
+          onSelect(itemKey, metaData);
+          this.setIsOpen(!this.isOpen);
+        });
       }
-      this.setIsOpen(!this.isOpen);
     };
   }
 
+  wrapOnClose() {
+    return (itemKey, event) => {
+      this.addButtonToggle = false;
+      this.setIsOpen(false);
+      let removedTabIndex = -1;
+      const updatedTabData = this.state.visibleTabData
+        .map((tab, index) => {
+          if (tab.itemKey === itemKey && tab.isSelected === true) {
+            removedTabIndex = index;
+          }
+          return { ...tab, originalIndex: index };
+        })
+        .filter((tab) => tab.itemKey !== itemKey);
+
+      if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled !== true) {
+        if (updatedTabData[removedTabIndex - 1].isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 1].isSelected = true;
+        }
+      } else if (!updatedTabData.some((tab) => tab.isSelected === true) && updatedTabData.length > 0 && removedTabIndex !== 0 && updatedTabData[removedTabIndex - 1] !== undefined && updatedTabData[removedTabIndex - 1].isDisabled === true) {
+        if (updatedTabData[removedTabIndex - 2]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex - 2].isSelected = true;
+        }
+      } else if (removedTabIndex >= 0 && removedTabIndex < updatedTabData.length) {
+        if (updatedTabData[removedTabIndex]?.isSelected !== undefined) {
+          updatedTabData[removedTabIndex].isSelected = true;
+        }
+      }
+      this.props.onTabStateChange(updatedTabData, itemKey, event);
+      this.setState({ visibleTabData: updatedTabData });
+    };
+  }
+
+  wrapOnAddButton() {
+    if (this.props.onSelectAddButton) {
+      this.addButtonToggle = true;
+      this.props.onSelectAddButton();
+    }
+  }
+
   render() {
     const {
-      ariaLabel, variant, onChange, isDraggable,
+      ariaLabel, variant, onChange, onSelectAddButton, ariaLabelAddTab, isDraggable,
     } = this.props;
     const theme = this.context;
     const enabledTabs = this.state.visibleTabData.filter(tab => !tab.isDisabled);
     const ids = enabledTabs.map(tab => tab.id);
     const hiddenIds = [];
     const visibleTabs = [];
     const hiddenTabs = [];
+    const moreIds = ids;
+    if (onSelectAddButton) { moreIds.push(this.addTabId); }
```

Comment:
```suggestion
    if (onSelectAddButton) { ids.push(this.addTabId); }
```
Curios !! does using `ids` array like above has any implications ?

Reply:
This moreIDs is being used only when AddTab functionality has been added in the Tabs.